crates/burn-compute/src/channel/base.rs

```diff
@@ -26,6 +26,9 @@ pub trait ComputeChannel<Server: ComputeServer>: Clone + core::fmt::Debug + Send
     /// Executes the `kernel` over the given `bindings`.
     fn execute(&self, kernel: Server::Kernel, bindings: Vec<Binding<Server>>);
 
+    /// Submit the current queued tasks to the server.
+    fn submit(&self);
```

Comment:
Do we want to have a submit function here or pass an option to the `sync` method? I'm unsure which one is the most intuitive. It's a bit weird to have an exposed `submit` function along `execute` and `sync`. 

```rust
fn sync(&self, options: SyncOptions);

#[derive(Clone, Copy)]
enum SyncOptions {
    /// Sync the execution queue.
    Queue,
    //// Sync the execution queue and wait for all tasks to be completed.
    Full,
}
```

Reply:
Ah alright! Yeah that might be more consistent.

It does slightly complicate things as for better or worse it kind of forces this to be exposed on the backend trait. Lmk how this looks!